src/Draco.Compiler/Internal/Syntax/Formatting/DracoFormatter.cs

```diff
@@ -0,0 +1,404 @@
+using System;
+using System.Linq;
+using Draco.Compiler.Api.Syntax;
+
+namespace Draco.Compiler.Internal.Syntax.Formatting;
+
+internal sealed class DracoFormatter : Api.Syntax.SyntaxVisitor
+{
+    private readonly FormatterSettings settings;
+    private FormatterEngine formatter = null!;
```

Comment:
Just passing allow settings in the ctor, no need to keep this mutable or init with `null!`